rholang/src/main/scala/coop/rchain/rholang/interpreter/rholangADT.scala

```diff
@@ -110,15 +160,25 @@ case class Send(chan: Channel, data: List[Par], persistent: Boolean)
 case class Receive(binds: List[(List[Channel], Channel)],
                    body: Par,
                    persistent: Boolean,
-                   count: Int)
+                   count: Int) {
+  override def toString: String =
+    (for { (xs, chan) <- binds; if binds.nonEmpty } yield {
+      xs.mkString(", ") + " <- " + chan.toString
+    }).mkString("for( ", " ; ", " )") + "{ " + body + " }"
+}
 
-case class Eval(channel: Channel)
+case class Eval(channel: Channel) {
+  override def toString: String = "*" + channel
+}
 
 // Number of variables bound in the new statement.
 // For normalized form, p should not contain solely another new.
 // Also for normalized form, the first use should be level+0, next use level+1
 // up to level+count for the last used variable.
-case class New(count: Int, p: Par)
+case class New(count: Int, p: Par) {
+  override def toString: String =
+    "new " + (0 until count).toList.mkString(", ") + " in { " + p.toString + " }"
```

Comment:
This suggests that you should either just do new:count as the pretty printing, or that pretty printing needs a level based counter so that it can print new xj, xj_+_1, xj_+_k_-_1.